app/SettingsScreen.js

```diff
@@ -0,0 +1,127 @@
+import React, {Component} from 'react';
+import { View, Text, Button, Switch, StyleSheet, Alert, TextInput, AsyncStorage } from 'react-native';
+
+export default class SettingsScreen extends Component{
+	state = {
+		blackBackground: false,
+		username: '',
+	}
+
+	componentDidMount(){
+		this.storageToState();
+	}
+
+	storageToState = async () => { 
+		try {
+  			const value = await AsyncStorage.getItem('@DemoRNStore:settings_background');
+  			if (value !== null){
+    			this.setState({blackBackground: value == 'true'});
+  			}
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert(JSON.stringify(error));
+		}
+		try {
+  			const value = await AsyncStorage.getItem('@DemoRNStore:settings_username');
+  			if (value !== null){
+    			this.setState({username: value});
+  			}
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('get username error');
+		}
+	}
+
+	toggleSwitch = (value) => {
+		this.setState({blackBackground: value});
+		this.storeBackground(value);
+	}
+
+	storeBackground = async (value) => {
+		try {
+  			await AsyncStorage.setItem('@DemoRNStore:settings_background', value.toString());
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('put background error');
+		}
+	}
+
+	onUserNameChanged = (text) => {
+		this.setState({username: text});
+		this.storeUserName(text);
+	}
+
+	storeUserName = async (username) => {
+		try {
+  			await AsyncStorage.setItem('@DemoRNStore:settings_username', username);
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('put username error');
+		}
+	}
+
+	containerStyle = () => {
+		let container = {
+			flex: 1,
+		};
+		if (this.state.blackBackground) {
+			container.backgroundColor = 'black';
+		}
+		return container;
+	}
+
+	render(){
+		let { blackBackground } = this.state;
+		return(
+		<View style={ this.containerStyle() }>
+			<View style={styles.row}>
+				<Text style={styles.leftItem} >Black background</Text>
+				<Switch
+					style={styles.rightItem}
+					onValueChange = { this.toggleSwitch }
+					value = {blackBackground}
+				/>
+			</View>
+			<View style={styles.row}>
+				<Text style={styles.leftItem} >Input your name</Text>
+				<TextInput
+        			style={styles.rightItem, {borderColor: 'gray', borderWidth: 1, width: 150}}
```

Comment:
Why pass two separate style objects? The right one is static styles anyway, could have put it in styles.rightItem. I am not sure your styles.rightItem work here anyway.